src/lib.rs

```diff
@@ -187,3 +187,71 @@ pub enum ReflinkSupport {
     /// Reflink support is unconfirmed.
     Unknown,
 }
+
+/// Creates a reflink of a specified block from one file to another.
+///
+/// This function is designed to be high performant and does not perform any extra API calls.
+/// It is expected that the user takes care of necessary preliminary checks and preparations.
+///
+/// If you need to clone an entire file, consider using the [`reflink`] or [`reflink_or_copy`]
+/// functions instead.
+///
+/// > Note: Currently the function works only for windows. It returns `Err` for any other platform.
+///
+/// # Windows Restrictions and Remarks
+/// - The source and destination regions must begin and end at a cluster boundary.
+/// - The cloned region must be less than 4GB in length.
```

Comment:
I believe that this restriction would make it harder to use cross-platform.

Perhaps we can add a loop to abstract over platform differences?